src/validators/union.rs

```diff
@@ -37,30 +39,49 @@ impl Validator for UnionValidator {
         extra: &Extra,
         slots: &'data [CombinedValidator],
     ) -> ValResult<'data, PyObject> {
-        // 1st pass: check if the value is an exact instance of one of the Union types
-        if let Some(res) = self
-            .choices
-            .iter()
-            .map(|validator| validator.validate_strict(py, input, extra, slots))
-            .find(ValResult::is_ok)
-        {
-            return res;
-        }
+        match self.strict {
```

Comment:
If/else might be more readable here?

Reply:
As you wish! You wrote match self.strict almost everytime (you used `if...else` only once 😆)